src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md

```diff
@@ -0,0 +1,222 @@
+
+```js
+import capitalize from 'lodash.capitalize';
+capitalize('hello'); // Hello
+```
+
+When the [`lodash.capitalize`](https://www.npmjs.com/package/lodash.capitalize) method is executed, it calls pre-written JavaScript code that capitalizes the first character of a string.
+
+### Example use of a JavaScript Framework
+
+A JavaScript framework is a predefined code template within which you construct your application's behavior. That is, when you use a framework, the framework controls the application flow. To use a framework, you write your custom application code, and then the framework calls your application code. 
+
+The following example shows a code snippet that uses the [preact](https://preactjs.com/) JavaScript framework:
+
+```js
+import { createElement } from 'preact';
+
+export default function App() {
+  return (
+    <p class="big">Hello World!</p>
+  )
+}
+```
+
+In the example, notice that the framework has a lot more control over the code you write, and in some cases, the framework even takes control over when to execute your code.
+
+## Why use a library?
+
+Using a JavaScript library can help to avoid unnecessary [code repetition](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Libraries can abstract away complex logic, such as date manipulation or financial calculations. A library can also help get your [initial product](https://en.wikipedia.org/wiki/Minimum_viable_product) out, rather than having to write all the code from scratch, which can take time.
+
+Some client-side JavaScript libraries help abstract away quirks of the web platform. A library can also serve as a learning tool. For example, if you're unfamiliar with animation easing [functions](https://easings.net/), the source code of a library can teach you how such easings work.
+
+Some libraries are backed by large companies that invest time and money into keeping libraries up to date and secure. Many libraries are accompanied by extensive documentation, which offers you and your team a quick way to familiarize yourself with the library's usage.
+
+Ultimately, using a JavaScript library saves you time.
+
+## Why should you care about library usage?
+
+Technically, you can develop your web application from scratch, but why go to the trouble when you can use free (open source) software, or purchase a solution that, in the long run, can save time and money? There are a large number of JavaScript libraries and frameworks available, each offering a unique approach to solving problems, and each having different characteristics. For example:
+
+-  A library can be written and maintained internally rather than by a third party.
+-  A library can have specific legal licenses that make it suitable or unsuitable for your web application.
+-  A library can be outdated or unmaintained.
+-  A library can simplify a set of complex tasks and save you a lot of time and money.
+-  A library can be widely used in the community, and can be well known amongst developers.
```

Comment:
```suggestion
Technically, you can develop your web application from scratch, but why go to the trouble when you can use free (open source) software, or purchase a solution that, in the long run, can save time and money? There are a large number of JavaScript libraries and frameworks available, each offering a unique approach to solving problems, and each having different characteristics. For example, a library can:

-  Be written and maintained internally rather than by a third party.
-  Have specific legal licenses that make it suitable or unsuitable for your web application.
-  Be outdated or unmaintained.
-  Simplify a set of complex tasks and save you a lot of time and money.
-  Be widely used in the community, and can be well known amongst developers.
```

Reply:
fyi the reason for the extra clarity/redundancy in my bullet points is to a) generally be more explicit/clear to the reader, b) maybe avoid stuff like https://twitter.com/soft/status/1449406390976409600 !

I don't mind if it needs to be changed though 👍

Reply:
Your call! I felt it flowed better this way, but your way is fine too! :)

Reply:
that tweet is interesting, probably something for writers to discuss, but I'll leave this as is.